sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -661,7 +666,9 @@ class Analyzer(
           val evaluatedOrderings = resolvedAliasedOrdering.zip(sortOrder).map {
             case (evaluated, order) =>
               val index = originalAggExprs.indexWhere {
-                case Alias(child, _) => child semanticEquals evaluated.child
+                case a@Alias(child, _) =>
+                  (child semanticEquals evaluated.child) ||
+                    a.exprId == evaluated.child.asInstanceOf[AttributeReference].exprId
                 case other => other semanticEquals evaluated.child
```

Comment:
We can put the `finalSortOrders` at the final place, after this `evaluatedOrderings`, it should be:
```
val sortOrdersMap = unresolvedSortOrders.map(new TreeNodeRef(_)).zip(evaluatedOrderings).toMap
val finalSortOrders = sortOrders.map(s => sortOrdersMap.getOrElse(TreeNodeRef(s), s))

if (sortOrder == finalSortOrders) {
......
```